src/jit/codegencommon.cpp

```diff
@@ -3942,7 +3942,8 @@ void            CodeGen::genFnPrologCalleeRegArgs(regNumber xtraReg,
                 
                 if ((!doingFloat &&
                     ((structDesc.eightByteClassifications[slotCounter] == SystemVClassificationTypeInteger) || 
-                     (structDesc.eightByteClassifications[slotCounter] == SystemVClassificationTypeIntegerReference))) ||
+                     (structDesc.eightByteClassifications[slotCounter] == SystemVClassificationTypeIntegerReference) ||
```

Comment:
This code is quite difficult to read. Can we add an abstraction: structDesc.IsIntegral(slotCounter) and structDesc.IsFloat(slotCounter)?